autogen/code_utils.py

```diff
@@ -214,12 +214,14 @@ def timeout_handler(signum, frame):
 
 
 def _cmd(lang):
-    if lang.startswith("python") or lang in ["bash", "sh", "powershell"]:
+    if lang.startswith("python") or lang in ["bash", "sh", "pwsh"]:
         return lang
     if lang in ["shell"]:
         return "sh"
     if lang in ["ps1"]:
-        return "powershell"
+        return "pwsh"
+    if lang in ["powershell"]:
```

Comment:
why not merge the conditions in line 221 and 223?
Can we add a simple test to cover the if branch?

Reply:
But then if lang = powershell we wont have any case to deal with that right? For the users that still use old powershell versions. Is there any other way we can cover that if branch with a test that I am unaware of at the moment?

Reply:
I mean changing line 221 into: `if lang in ["ps1", "powershell"]:`

Reply:
Ohh okay got it yes you're right that would be more efficient, sorry I'll make the changes right away